### ISSUE TITLE
"copy markdown link" is replacing `[]` and `()` with underscores incorrectly

### ISSUE THREAD

**Issue by mgrandi** [laurent22/joplin#1244]
### Operating system

- macOS

### Application
- Desktop

### Version
Joplin 1.0.127 (prod, darwin)

# Description
When you have a note title that has some special characters, it seems that when you `right click -> copy markdown link`, the generated markdown link snippet seems to be escaping some characters in the title portion of the link with underscores, which means the title is italicized in the link, when that is not necessarily what is desired

This escaping is probably happening because the markdown link syntax is of the form `[title](link)`, which has both parenthesis and brackets, but if you see the example below, if you manually replace the escaped `_` with the correct character (either `( ` or `[` ), it still works, so I question why this escaping exists in the first place

# Examples:

Note 1: a note with `testing note [brackets]` as the title
Note 2: a note with `testing note (with some parens)` as the title
Note 3: a new note, just for pasting the markdown links into

right click Note 1, copy markdown link, paste into Note 3
right click Note 2, copy markdown link, paste into Note 3

contents of Note 3:

```
[testing note _with some parens_](:/3acb011934bd4c8bbf83c9908e02d1e4)

[testing note _brackets_](:/6c3d4cc77bc04d278605f2f8ee68208f)
```

Expected contents of note 3: 

```
[testing note (with some parens)](:/3acb011934bd4c8bbf83c9908e02d1e4)

[testing note [brackets]](:/6c3d4cc77bc04d278605f2f8ee68208f)
```

Screenshot: 

![screenshot_1149](https://user-images.githubusercontent.com/802786/53051906-96032400-345a-11e9-8e3c-03ad6f8698c3.png)

(you can note that the syntax highlighting seems to be broken for the 4th entry, where i replaced `_` with the brackets, but clicking the link still works)

**Comment by foxmask** [comment]
did you make upgrade and new test ? 
because I made a lot of notes with "copy markdown link" to build a note which is just an index of other pages without any issues.

**Comment by mgrandi** [comment]
This is still happening with the latest version: `Joplin 1.0.160 (prod, darwin)`

i have a note called `2019-06-13 - PMD-1879 Notes (liquibase defaults ticket)`, but when i right click -> "copy markdown link" and then paste it into a new note, it pastes: 

```
[2019-06-13 - PMD-1879 Notes _liquibase defaults ticket_](:/8310c1cec2f641a7b1bcb29a961b307f)
```

**Comment by foxmask** [comment]
But it is the expected value for me. Not for you ?

**Comment by mgrandi** [comment]
i would expect the link text to match the actual name of the note, but Joplin is incorrectly replacing `[`, `]`, `(`, and `)` with `_` so it is partially italicized

**Comment by mgrandi** [comment]
this is still an issue yes

**Comment by philtrade** [comment]
Seems being handled [ #2233 ](https://github.com/laurent22/joplin/pull/2233)

**Comment by mgrandi** [comment]
thanks for fixing it!

**Comment by mgrandi** [comment]
looks like this is solved in at least the version i'm using (1.0.197)

a testing note with the title `Testing Joplin Note [test] (test)` has the output when right clicked->copy markdown link : `[Testing Joplin Note \[test\] (test)](:/e9f8ddaf06254055ae24a4b6679394fc)`

closing :)
